.github/workflows/github-actions.yml

```diff
@@ -17,9 +17,15 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install and configure Poetry
         uses: snok/install-poetry@v1
-      - name: Install dependencies
+      - name: Install all dependencies
         run: |
           poetry install --all-extras
       - name: Run tests
         run: |
           poetry run pytest
+      - name: Install basic dependencies
+        run: |
+          poetry install
```

Comment:
Why install all dependencies, too? Later, install “basic” dependencies (which are already included in the “all” ones)?

Reply:
Perhaps I should find a better name and invert thte execution process....
So, first install the "basic" dependencies dosen't install pandas and geopandas, to run the test without those packages.

Then, install the dependencies with extras, which means, pandas and geopandas...
What do yout think?

Reply:
> invert thte execution process....

I think you need two separate jobs/processes on the CI. Maybe three. One without Pandas or GeoPandas, one with Pandas (but not GeoPandas), and another one with GeoPandas.

Reply:
>I think you need two separate and independent jobs/processes on the CI. Maybe three. One without Pandas or GeoPandas, one with Pandas (but not GeoPandas), and another one with GeoPandas.
I agree. I will revert the change I have done and add those jobs on a different PR.
ok?

Reply:
Have you manually tested executing the library without Pandas and without GeoPandas? Otehrwise we need it in this PR.

Reply:
Yes I have tested manually without pandas and geopandas, with pandas only, and with pandas and geopandas.

Reply:
Ok. But keeping [the logic that arguably does not make sense](https://github.com/FelipeSBarros/crossfire/pull/103/files#r1531416332)?

Reply:
@cuducos you made me thought about the importance of having this issue solved in this PR.
Because of that, Ihave decided to commit an proposal that try yo solve the #104 ;

Reading about [github flow](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow) I saw that e can have as many jobs as it is needed.
In our case, I note at least three is needed:
    1. one to test several python versions with the basic dependencies (i.e. without Pandas and GeoPandas); 
    2. another to test several python versions with Pandas dependencies only and, 
    3. the last one to test several python versions with both Pandas and GeoPandas dependencies.

Trying to make it more organized, I set the prefix `(basic dependencies)`, `(Pandas only)` and `(Pandas & GeoPandas)` before the Python version information and, also, I set [prerequisites](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#example-requiring-successful-dependent-jobs) between jobs, so we first have the test run with basic dependencies; once finished, it will run the test with Pandas only and, then, if passing, the test with both Pandas and GeoPandas.

I understand that perhaps this is a misusing of this feature, and that making one set of tests require onthers to pass make it take longer. But I like the way it keeps thinks more organized...

![image](https://github.com/FelipeSBarros/crossfire/assets/7208674/36f413a0-d4ab-495c-b474-aeecefb7915b)